### PR TITLE
PoS Add Bitset to messaging schema

### PR DESCRIPTION
**Pull request by tholonious**

### CODE REVIEW

**Comment by tholonious** [comment]
Current dependencies on/for this PR:
* feature/pos-messaging-schema
  * **PR #555** <a href="https://app.graphite.dev/github/pr/deso-protocol/core/555" target="_blank">Graphite</a> 
    * **PR #558** <a href="https://app.graphite.dev/github/pr/deso-protocol/core/558" target="_blank">Graphite</a> 
      * **PR #564** <a href="https://app.graphite.dev/github/pr/deso-protocol/core/564" target="_blank">Graphite</a> 
        * **PR #565** <a href="https://app.graphite.dev/github/pr/deso-protocol/core/565" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/deso-protocol/core/565?utm_source=stack-comment).

**Review by lazynina — APPROVED** [review]

File: lib/pos_network.go [hunk 1]
```diff
@@ -365,15 +364,21 @@ func (sig *AggregatedBLSSignature) Eq(other *AggregatedBLSSignature) bool {
...
 
 func (sig *AggregatedBLSSignature) ToBytes() ([]byte, error) {
 	retBytes := []byte{}
 
 	// SignersList
-	retBytes = append(retBytes, EncodeByteArray(sig.SignersList)...)
+	if sig.SignersList == nil {
+		return nil, errors.New("AggregatedBLSSignature.ToBytes: SignersList must not be nil")
+	}
+	retBytes = append(retBytes, EncodeByteArray(sig.SignersList.ToBytes())...)
```
Comment (AeonSw4n) [inline lib/pos_network.go:381]: I had a related comment for #564, which would remove the need for `EncodeByteArray`, but this route is fine too.

**Review by AeonSw4n — APPROVED** [review]
Very chill

File: lib/pos_network.go (hunk 1 above)
Comment (tholonious) [reply lib/pos_network.go:381]: Good point. I do think that the `lib` package should be responsible for encoding the bitlist for network messages, but you're right that using the raw EncodeByteArray here is non-ideal.

I added a new `EncodeBitset` and `DecodeBitset` functions to wrap the byte encoding and do the length prefixing.

**Pull request merged**